From cfecdb53470d3d48449c170fa55188ed0b4bb055 Mon Sep 17 00:00:00 2001
From: TFLite Patcher <tflite@localhost>
Date: Fri, 18 Oct 2024 18:11:43 +0800
Subject: [PATCH] Add system lib BUILD rules for absl/log

The linkopts were generated by
$ pkg-config-brya --libs-only-l absl_check absl_log

PATCH_NAME=absl-log
---
 third_party/absl/system.absl.log.BUILD | 52 ++++++++++++++++++++++++++
 third_party/absl/workspace.bzl         |  1 +
 2 files changed, 53 insertions(+)
 create mode 100644 third_party/absl/system.absl.log.BUILD

diff --git a/third_party/absl/system.absl.log.BUILD b/third_party/absl/system.absl.log.BUILD
new file mode 100644
index 00000000..d14be553
--- /dev/null
+++ b/third_party/absl/system.absl.log.BUILD
@@ -0,0 +1,52 @@
+load("@rules_cc//cc:defs.bzl", "cc_library")
+
+package(default_visibility = ["//visibility:public"])
+
+[cc_library(
+    name = n,
+    linkopts = [
+        "-labsl_bad_optional_access",
+        "-labsl_bad_variant_access",
+        "-labsl_base",
+        "-labsl_city",
+        "-labsl_civil_time",
+        "-labsl_debugging_internal",
+        "-labsl_demangle_internal",
+        "-labsl_examine_stack",
+        "-labsl_graphcycles_internal",
+        "-labsl_hash",
+        "-labsl_int128",
+        "-labsl_kernel_timeout_internal",
+        "-labsl_log_entry",
+        "-labsl_log_globals",
+        "-labsl_log_internal_conditions",
+        "-labsl_log_internal_format",
+        "-labsl_log_internal_globals",
+        "-labsl_log_internal_log_sink_set",
+        "-labsl_log_internal_message",
+        "-labsl_log_internal_nullguard",
+        "-labsl_log_internal_proto",
+        "-labsl_log_severity",
+        "-labsl_log_sink",
+        "-labsl_low_level_hash",
+        "-labsl_malloc_internal",
+        "-labsl_raw_logging_internal",
+        "-labsl_spinlock_wait",
+        "-labsl_stacktrace",
+        "-labsl_str_format_internal",
+        "-labsl_strerror",
+        "-labsl_string_view",
+        "-labsl_strings",
+        "-labsl_strings_internal",
+        "-labsl_symbolize",
+        "-labsl_synchronization",
+        "-labsl_throw_delegate",
+        "-labsl_time",
+        "-labsl_time_zone",
+    ],
+) for n in [
+    "absl_check",
+    "absl_log",
+    "check",
+    "log",
+]]
diff --git a/third_party/absl/workspace.bzl b/third_party/absl/workspace.bzl
index 9565a82c..addb2410 100644
--- a/third_party/absl/workspace.bzl
+++ b/third_party/absl/workspace.bzl
@@ -20,6 +20,7 @@ def repo():
         "flags",
         "functional",
         "hash",
+        "log",
         "memory",
         "meta",
         "numeric",
